src/main/java/ulcrs/controllers/UlcController.java

```diff
@@ -0,0 +1,64 @@
+package ulcrs.controllers;
+
+import spark.Request;
+import spark.Response;
+import spark.RouteGroup;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLConnection;
+
+import static spark.Spark.before;
+import static spark.Spark.get;
+
+public class UlcController extends BaseController {
+
+    @Override
+    public RouteGroup routes() {
+        return () -> {
+            before("/*", (request, response) -> log.info("endpoint: " + request.pathInfo()));
+            get("/login", this::loginTest);
+        };
+    }
+
+    private String loginTest(Request request, Response response) {
+
+        URL url;
+        URLConnection urlConn;
+        BufferedReader br = null;
+        String line;
+
+        StringBuilder page = new StringBuilder();
+
+        try {
+            url = new URL("https://tbrtest.engr.wisc.edu/");
+            urlConn = url.openConnection();
+
+            // insert cookie here
+            urlConn.setRequestProperty("Cookie", "<cookie name>=<cookie>");
```

Comment:
Same for this as well. You could use a sprintf-like function to fill this in with constants at the top of the class.

Reply:
ditto from my other comment